src/setuptools_pyproject_migration/__init__.py

```diff
@@ -91,8 +91,67 @@ def _generate(self) -> Pyproject:
         if dependencies:
             pyproject["project"]["dependencies"] = dependencies
 
+        entrypoints = self._generate_entrypoints(dist.entry_points)
+
+        # GUI scripts and console scripts go separate in dedicated locations.
+        if "console_scripts" in entrypoints:
+            pyproject["project"]["scripts"] = entrypoints.pop("console_scripts")
+
+        if "gui_scripts" in entrypoints:
+            pyproject["project"]["gui-scripts"] = entrypoints.pop("gui_scripts")
+
+        # Anything left over gets put in entry-points
+        if entrypoints:
+            pyproject["project"]["entry-points"] = entrypoints
+
         return pyproject
 
+    @staticmethod
+    def _parse_entrypoint(entrypoint: str) -> Tuple[str, str]:
+        """
+        Extract the entry point and name from the string.
+
+        .. code-block:: python
+
+            WritePyproject._parse_entrypoint("hello-world = timmins:hello_world")
+            # returns ('hello-world', 'timmins:hello_world')
+
+            WritePyproject._parse_entrypoint("hello-world")
+            # throws ValueError: Entry point 'hello-world' is not of the form 'name = target'
+
+        :param: entrypoint  The entry point string, of the form
+                            "entrypoint = module:function" (whitespace optional)
+        :returns:           A two-element `tuple`, first element is the entry point name, second element is the target
+                            (module and function name) as a string.
+        :raises ValueError: An equals (`=`) character was not present in the entry point string.
+        """
+        # Format: 'hello-world = timmins:hello_world'
+        if "=" not in entrypoint:
+            raise ValueError("Entry point %r is not of the form 'name = module:function'" % entrypoint)
+
+        (name, target) = entrypoint.split("=", 1)
+        return (name.strip(), target.strip())
+
+    @classmethod
+    def _generate_entrypoints(cls, entrypoints: Optional[Dict[str, List[str]]]) -> Dict[str, Dict[str, str]]:
+        """
+        Dump the entry-points given, if any.
+
+        :param: entrypoints The `entry_points` property from the
+                            :py:class:setuptools.dist.Distribution being examined.
+        :returns:           The entry points, split up as per
+                            :py:meth:_parse_entrypoint and grouped by entry point type.
+        """
+        if not entrypoints:
+            return {}
+
+        parsedentrypoints: Dict[str, Dict[str, str]] = {}
+
+        for eptype, raweps in entrypoints.items():
+            parsedentrypoints[eptype] = dict(map(cls._parse_entrypoint, raweps))
+
+        return parsedentrypoints
```

Comment:
Definitely optional, but I think this helps with readability (and plus it's consistent with the way we separate `_parse_entrypoint`):

```suggestion
        parsed_entrypoints: Dict[str, Dict[str, str]] = {}

        for eptype, raweps in entrypoints.items():
            parsed_entrypoints[eptype] = dict(map(cls._parse_entrypoint, raweps))

        return parsed_entrypoints
```

Or, I guess what we really _should_ do is make it consistent with the way we spell "entrypoints"/"entry-points"/"entry points" :shrug:

Reply:
True, I vaguely recall there being conventions around using underscores in variable names… but seems it's on _packages_ where underscores are discouraged, not variables.

https://peps.python.org/pep-0008/#package-and-module-names

So yeah, when I next get a moment… a underscore should be added here.